### ISSUE TITLE
Show warning when connection to an outdated API version

### ISSUE THREAD

**Issue by andreaswe** [cloudfoundry/cli#1088]
## Feature Request
Scenario:
- Old CF version installed (with an outdated API version)
- User connects to API using the most recent CLI
Now the CLI always shows the help for the most recent API version.
This can lead to confusion when the user tries to use an API command that is not supported by the installed CF API version (CLI shows help for most recent API, but user cannot execute some commands, since they are not supported by the API version the user connects to).

Here, the user should be informed that the CLI API version and the version of the CF API he connects to do not match.
Maybe the CLI could even point to the online documentation for the API version the user connects to.

**Comment by dkoper** [comment]
When you run a command (/flag/etc.) not supported by the CF release you are targeting, you (should) get a clear error message. E.g.:
```
diesk@cloud-cf:~/workspace$ ./cf tasks
FAILED
Unexpected response Error 404
Note that this command requires CF API version 3.0.0+.
```

I am not sure pointing CLI users to API documentation helps.
Also note there is no single API documentation site (the CLI calls APIs on the Cloud Controller, UAA, Routing, Loggregator and ssh endpoints).

We chose not to add minimum versions (for each API?) to help text - most commands should work fine so that would just clutter help.
Ideally CF operators keep up to date, or advise their users what CLI version to use, or list restrictions of their CF deployment.

Regards,
Dies Koper
CF CLI PM

**Comment by dkoper** [comment]
Discussed on slack with @jtuchscherer that the confusion came from running the following command against a CF target endpoint that did not support the "http" health check type:

```
$ ./cf set-health-check cf-docs http
Updating health check type for app (...)
The app is invalid: health_check_type must be one of port, none
FAILED
```

Will look into changing that message into something like:
```
Your target CF API version only supports health check types port, none.
```

**Comment by dkoper** [comment]
Resolved in 6.27.0. Thanks for reporting.
https://github.com/cloudfoundry/cli#downloads

Cheers,
Dies Koper
CF CLI PM
